text/0119-instance-options.md

```diff
@@ -0,0 +1,459 @@
+```
+
+The expected behavior of each flag is described in the list above.
+
+### 3.3 Exhaustive list of options
+
+- [Database path](#database-path)
+- [Environment](#environment)
+- [HTTP address & port binding](#http-address-port-binding)
+- [Master key](#master-key)
+- [Disable analytics](#disable-analytics)
+- [Dumps](#dumps-destination)
+- [Dumps destination](#dumps-destination)
+- [Import dump](#import-dump)
+- [Ignore missing dump](#ignore-missing-dump)
+- [Ignore dump if DB exists](#ignore-dump-if-db-exists)
+- [Log level](#log-level)
+- [Max index size](#max-index-size)
+- [Max TASK_DB size](#max-task-db-size)
+- [Payload limit size](#payload-limit-size)
+- [Snapshots](#schedule-snapshot-creation)
+- [Schedule snapshot creation](#schedule-snapshot-creation)
+- [Snapshot destination](#snapshot-destination)
+- [Snapshot interval](#snapshot-interval)
+- [Import snapshot](#import-snapshot)
+- [Ignore missing snapshot](#ignore-missing-snapshot)
+- [Ignore snapshot if DB exists](#ignore-snapshot-if-db-exists)
+- [Memory usage](#memory-usage)
+- [Indexing jobs](#indexing-jobs)
+- [SSL configuration](#ssl-authentication-path)
+- [SSL authentication path](#ssl-authentication-path)
+- [SSL certificates path](#ssl-certificates-path)
+- [SSL key path](#ssl-key-path)
+- [SSL OCSP path](#ssl-ocsp-path)
+- [SSL require auth](#ssl-require-auth)
+- [SSL resumption](#ssl-resumption)
+- [SSL tickets](#ssl-tickets)
+
+#### Database path
+
+**Environment variable**: `MEILI_DB_PATH`
+**CLI option**: `--db-path`
+**Default value**: `"data.ms/"`
+**Expected value**: a filepath
+
+Designates the location where database files will be created and retrieved.
+
+#### Environment
+
+**Environment variable**: `MEILI_ENV`
+**CLI option**: `--env`
+**Default value**: `development`
+**Expected value**: `production` or `development`
+
+Configures the instance's environment. Value must be either `production` or `development`.
+
+`production`:
+
+- Setting a master key is **mandatory**
+- The search preview interface is disabled
+
+`development`:
+
+- Setting a master key is **optional**
+- Search preview is enabled
+
+#### HTTP address & port binding
+
+**Environment variable**: `MEILI_HTTP_ADDR`
+**CLI option**: `--http-addr`
+**Default value**: `"127.0.0.1:7700"`
+**Expected value**: an HTTP address and port
+
+Sets the HTTP address and port Meilisearch will use.
+
+#### Master key
+
+**Environment variable**: `MEILI_MASTER_KEY`
+**CLI option**: `--master-key`
+**Default value**: `None`
+**Expected value**: an alphanumeric string
+
+Sets the instance's master key, automatically protecting all routes except [`GET /health`](/reference/api/health.md). This means you will need an API key to access endpoints such as `POST /search` and `GET /documents`.
+
+You must supply an alphanumeric string when using this option.
+
+Providing a master key is mandatory when `--env` is set to `production`; if none is given, then Meilisearch will throw an error and refuse to launch.
```

Comment:
Do we have some kind of validation in this option? Like: the `master-key` should have 10 characters at least to work?

Reply:
Nope, no validation, but can be considered in the future (it will be annoying for the integration team that currently run all the test with `masterKey` as master key 😂)

Reply:
YEAAAH! hahah 

Could be only required for the `production` env by the way :)